src/server/decoders.ts

```diff
@@ -20,16 +15,29 @@ interface ThriftDecoder<A> {
 
 // ----- Functions ----- //
 
-const decodeContent = <A>(decoder: ThriftDecoder<A>) => (content: Buffer | undefined): A => {
-    const transport = new BufferedTransport(content);
-    const protocol = new CompactProtocol(transport);
+async function toTransport(buffer: Buffer): Promise<TTransport> {
+    return new Promise((resolve, reject) => {
+        const writer = TBufferedTransport.receiver((transport, seqID) => {
+            resolve(transport)
+        }, 0);
+        writer(buffer);
+    });
+}
+
+const decodeContent = <A>(decoder: ThriftDecoder<A>) => async (content: Buffer | undefined): Promise<A> => {
+    if (content) {
+        const transport = await toTransport(content);
+        const protocol = new TCompactProtocol(transport);
 
-    return decoder.read(protocol);
+        return decoder.read(protocol);
+    } else {
+        return Promise.reject("Invalid request")
+    }
 }
 
-const capiDecoder = decodeContent(CapiResponse);
-const errorDecoder = decodeContent(ErrorResponse);
-const mapiDecoder = decodeContent(MapiContent);
+const capiDecoder = decodeContent(ItemResponseSerde);
```

Comment:
Serde?

Reply:
It stands for "SERialisation DEserialisation"

Like Modem: MOdulator DEModulator
Or Codec: COder DECoder